src/rollup/widgets/init-widget-array.js

```diff
@@ -97,33 +50,26 @@ export default function initializeWidgetArray (array) {
       }
     }
 
-    if (widget.type === 'message' && (widget.recommend && Object.keys(widget.recommend).length !== 0) || (widget.content && widget.content.length !== 0)) {
-      if (widget.layout !== 'slideout' && widget.layout !== 'modal' && widget.layout !== 'inline') {
-        throw new Error('Unsupported layout for content recommendation');
-      }
-
-      if (widget.content && widget.content[0] && !widget.content[0].default) {
-        throw new Error('Cannot define recommended content unless it is a default');
-      }
-
-      var params = widget.recommend;
-
-      if (params && params.collection) {
-        params.contentsegment = widget.recommend.collection;
-        delete params.collection;
-      }
-
-      recContent(widget, params);
+    preloadLio(widget, pf, function () {
+      preloadRecommendation(widget, pf, function () {
+        pf.initializeWidget(widget, options);
+        if (options && options.priority === 'ordered') {
```

Comment:
could have missed it but doesn't look like we validate that the `options.priority` outside of basically saying if its n e thing other than `ordered` treat it as default. eventually, it might be worth ensuring that if a priority is passed its valid.